test/test_nestedtensor.py

```diff
@@ -0,0 +1,173 @@
+# Owner(s): ["module: nestedtensor"]
+
+import torch
+from torch.testing._internal.common_utils import TestCase, load_tests
+from torch import nested_tensor
+
+# Tests are ported from pytorch/nestedtensor.
+# This makes porting as_nested_tensor easier in the future.
+def _iter_constructors():
+    # yield as_nested_tensor
+    yield nested_tensor
+
+
+class TestNestedTensor(TestCase):
+    @torch.inference_mode()
```

Comment:
Is this required?

Reply:
Yes, it is. NestedTensor does not support autograd. If omitted we run into strange errors (because composites that can't deal with NestedTensor are being called). We can register a fallback under AutogradNestedTensor that throws an exception if NestedTensor is called outside this context. Let's address this in a follow-up PR.

Reply:
OK, this is fine to follow up, but we definitely should fix this; the system should be capable of handling this case correctly.